compiler/src/dotty/tools/dotc/transform/OverridingPairs.scala

```diff
@@ -64,11 +64,25 @@ object OverridingPairs {
       decls
     }
 
+    /** Is `parent` a qualified sub-parent of `bc`?
+     *  @pre `parent` is a parent class of `base` and it derives from `bc`.
+     *  @return true if the `bc`-basetype of the parent's type is the same as
+     *               the `bc`-basetype of base. In that case, overriding checks
+     *               relative to `parent` already subsume overriding checks
+     *               relative to `base`. See neg/11719a.scala for where this makes
+     *               a difference.
+     */
+    protected def isSubParent(parent: Symbol, bc: Symbol)(using Context) =
+      bc.typeParams.isEmpty
+      || self.baseType(parent).baseType(bc) == self.baseType(bc)
```

Comment:
While this fixes the immediate problem, it's worth noting that Scala 2 completely removed the subparent logic in https://github.com/scala/scala/pull/7439/commits/cc55bd92dd53fe2eaff29943a3479782bda2ddf2 because:
> subclassing on traits does not imply one's linearisation is contained in the other's

The testcase from that commit is:
```scala
trait IO {
  def c(x: Int): Int = ???
}
trait SO extends IO {
  override final def c(x: Int): Int = ???
}
trait SOIO extends IO {
  override def c(x: Int): Int = ???
}
trait SOSO extends SOIO with SO
abstract class AS extends SO
class L extends AS with SOSO // error expected: c definined in SOIO overrides final method c in SO 
```
Which currently compiles in Dotty, even though we end up overriding a final def with a non-final def. We already have an issue open to keep track of this (https://github.com/lampepfl/dotty/issues/7551) so we don't necessarily need to handle this now.

Reply:
OK. We should evaluate the performance implications of subParents before removing them. Also bridge generation currently relies on the check being present, otherwise you would get duplicate bridges.

Reply:
> Also bridge generation currently relies on the check being present, otherwise you would get duplicate bridges.

Indeed, but Scala 2 had the same issue so https://github.com/scala/scala/commit/cc55bd92dd53fe2eaff29943a3479782bda2ddf2 also takes care of that:
```scala
// Skip nextEntry if the (non-trait) class in `parents` is a subclass of the owners of both low and high.
// this means we'll come back to this entry when we consider `nonTraitParent`, and the linearisation order will be the same
// (this only works for non-trait classes, since subclassing on traits does not imply one's linearisation is contained in the other's)
// This is not just an optimization -- bridge generation relies on visiting each such class only once.
if (!isMatch || (nonTraitParent.isNonBottomSubClass(low.owner) && nonTraitParent.isNonBottomSubClass(high.owner)))
```